presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/decoders/TimestampDecoder.java

```diff
@@ -16,24 +16,66 @@
 import io.prestosql.spi.PrestoException;
 import io.prestosql.spi.block.BlockBuilder;
 import io.prestosql.spi.connector.ConnectorSession;
-import org.elasticsearch.common.document.DocumentField;
 import org.elasticsearch.search.SearchHit;
 
+import java.time.DateTimeException;
 import java.time.Instant;
+import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.ResolverStyle;
+import java.time.format.SignStyle;
+import java.time.temporal.TemporalAccessor;
+import java.util.Locale;
+import java.util.Map;
 import java.util.function.Supplier;
 
 import static io.prestosql.spi.StandardErrorCode.NOT_SUPPORTED;
-import static io.prestosql.spi.StandardErrorCode.TYPE_MISMATCH;
 import static io.prestosql.spi.type.TimestampType.TIMESTAMP;
 import static java.lang.String.format;
-import static java.time.format.DateTimeFormatter.ISO_DATE_TIME;
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE;
+import static java.time.temporal.ChronoField.HOUR_OF_DAY;
+import static java.time.temporal.ChronoField.MINUTE_OF_HOUR;
+import static java.time.temporal.ChronoField.NANO_OF_SECOND;
+import static java.time.temporal.ChronoField.SECOND_OF_MINUTE;
 
 public class TimestampDecoder
         implements Decoder
 {
     private static final ZoneId ZULU = ZoneId.of("Z");
+    private static final DateTimeFormatter STRICT_DATE_OPTIONAL_TIME_FORMATTER = new DateTimeFormatterBuilder()
```

Comment:
Is this format going to be sufficient? We may need to extend the code to be able to parse a number of the supported formats. See https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-date-format.html

Reply:
This was actually modified from the elasticsearch Formatter.

The timestamps that I was getting back in the tests were either date fields (when calling .getFields() would automatically add a "start of day" time at 0 hours, 0 min, etc.. So this will at minimum require a date and optionally parse a timestamp. Then I handle the parsing by first trying a LocalDateTime and if that fails, just a LocalDate and  convert it to a DateTime.

My goal was mainly to support the minimum of what we see in the tests for now but extending the timestamp to handle date. Expanding to handle all formats will be a bit of work.

Reply:
Removed all this since using getter.get() as an alternative is only used for the nested case and all other fields use getFields() as before.